pnpm-lock.yaml

```diff
@@ -21786,7 +21786,7 @@ packages:
   /@types/concat-stream/1.6.1:
     resolution: {integrity: sha512-eHE4cQPoj6ngxBZMvVf6Hw7Mh4jMW4U9lpGmS5GBPB9RYxlFg+CHaVN7ErNY4W9XfLIEn20b4VDYaIrbq0q4uA==}
     dependencies:
-      '@types/node': 18.19.1
```

Comment:
This file looks unrelated to your change.

Reply:
If I do "nmp i" on clean repo, that's what I get. I'm not sure what I'm supposed to do here, to be honest. Not check in lock file? Manually try to undo everything it changes that seems unrelated?

Reply:
On a clean repo you should never do `pnpm i`. You should only ever do `pnpm i --frozen-lockfile`. You can do `pnpm i` after making a manual change to package.json dependencies (assuming you first did `pnpm i --frozen-lockfile` without those package dep changes).

Reply:
@jason-ha, to be honest, it's hard for me to figure out what one can do here, other than manually undo these changes (which sucks).

I'm not sure what context you put into "clean repo". I have a dozen branches that I cycle through on a single repo, and I have no clue (and do not track) their baseline commits in main (and when was the last time I pulled origin/main). So, whenever I switch between branches, and do (optionally) some coding on top of it, I need to run something to install and build it. I never have "clean repo" state as I never cycle through main (except if I start new project).

The workaround that exists (for cases where I just merged origin/main into the branch I'm working on) - sync another fork to roughly same commit in main, do "pnpm i --frozen-lockfile" on that clean fork, copy lock file to a fork with my changes, and then run npm i there. That used to work for me, but right now (for some reason) I'm getting the following error when runing "pnpm i --frozen-lockfile", but not if I run "npm i"!

```
Lockfile is up to date, resolution step is skipped
 ERR_PNPM_UNSUPPORTED_ENGINE  Unsupported environment (bad pnpm and/or Node.js version)

Your Node version is incompatible with "/jwt-decode/4.0.0".

Expected version: >=18
Got: v14.19.2
```

If I run node -v, I get v20.11.0, and I do not have any node managers installed.
And as I mentioned, "npm i" works on that repo with no problems, but it modifies lock file :(

It's pretty shitty place to be, to be honest.